apis/v1alpha1/gateway_types.go

```diff
@@ -302,6 +297,8 @@ const (
 // * "UDP"
 //
 // +kubebuilder:validation:Enum=HTTP;HTTPS;TLS;TCP;UDP
+// +kubebuilder:validation:MaxLength=1
+// +kubebuilder:validation:MaxLength=100
```

Comment:
Length validation are redundant due to the enum.

Reply:
Good call, I think we'll have to add this back in whenever we add more flexible protocol support but that's not in scope here. https://github.com/kubernetes-sigs/service-apis/issues/203